include/seqan/index/find_swift.h

```diff
@@ -1017,14 +1017,24 @@ inline void _createHit(
 
     int64_t ndlBegin = lastInc + length(pattern.shape) - diag - height;
     auto const & queryInfix = seqan2::getSequenceByNo(ndlSeqNo, seqan2::indexText(seqan2::needle(pattern)));
-    // can not extend seed past the beginning of the host sequence
+    
     if (ndlBegin + (int64_t) seqan2::beginPosition(queryInfix) < 0LL)
-        return;
+    {
+        std::clamp(ndlBegin, -(int64_t)seqan2::beginPosition(queryInfix), (int64_t) 0);
+        height = lastInc + length(pattern.shape) - diag - ndlBegin;
+        // can not extend seed past the beginning of the host sequence
+        if (height < width)
```

Comment:
The comment seems to be misplaced? Possibly on top of the surrending if clause?

Reply:
No, the seed can not be extended past the bounds of the **host sequence**, so I return without creating a SWIFT hit. The if statement is extending the seed outside the bounds of the query **segment**

Reply:
So the width being greater than the height implies that the seed cannot be extended past the host sequences? Maybe you can add a short comment on what height and width means in this context. Just for future references, when someone else is stumbling over this.